LdapForNet/Native/LdapNativeWindows.cs

```diff
@@ -33,6 +33,12 @@ internal override int TrustAllCertificates(SafeHandle ld)
                 Marshal.GetFunctionPointerForDelegate<VERIFYSERVERCERT>((connection, serverCert) => true));
         }
 
+        internal override int TrustAllCertificatesTls(SafeHandle ld)
+        {
+            return ldap_set_option(ld, (int)Native.LdapOption.LDAP_OPT_SERVER_CERTIFICATE,
```

Comment:
What difference between TrustAllCertificatesTls and just TrustAllCertificates ?

Reply:
If I'm understanding the documentation correctly, nothing needs to be changed. TrustAllCertificatesTls() will be called when you call StartTransportLayerSecurity(true). Before, when you called StartTransportLayerSecurity(true), it would in turn call TrustAllCertificates().

The only change between TrustAllCertificatesTls() and TrustAllCertificates() is that on Windows, TrustAllCertificates() forces you to use SSL/TLS (LDAPS). That works great for an SSL/TLS certificate, but not for a STARTTLS certificate.

Reply:
So could you add the default parameter fo TrustAllCertificates ``` CerificateType = CertificateType.Ssl ``` and resolve your new logic in TrustAllCertificates if CertificateType.StartTls ?

Reply:
Yes, that should work. Let me change that quick.